bigquery/google/cloud/bigquery/table.py

```diff
@@ -741,11 +741,9 @@ def insert_data(self,
             row_info = {}
 
             for field, value in zip(self._schema, row):
-                if field.field_type == 'TIMESTAMP':
-                    # BigQuery stores TIMESTAMP data internally as a
-                    # UNIX timestamp with microsecond precision.
-                    # Specifies the number of seconds since the epoch.
-                    value = _convert_timestamp(value)
+                converter = _SCALAR_VALUE_TO_JSON_ROW.get(field.field_type)
+                if converter is not None:
```

Comment:
Sorry I didn't acknowledge that before. I understand it'll be faster, e.g.

```
In [1]: def f(x):
   ...:     return x
   ...: 

In [2]: %timeit f(10)
The slowest run took 29.10 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 65.5 ns per loop

In [3]: def g1(x):
   ...:     return f(x)
   ...: 

In [4]: def g2(x):
   ...:     if x:
   ...:         return x
   ...: 

In [5]: %timeit g1(10)
The slowest run took 114.21 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 123 ns per loop

In [6]: %timeit g2(10)
The slowest run took 26.05 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 73.2 ns per loop

In [7]: %timeit g1(10)
The slowest run took 40.91 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 122 ns per loop

In [8]: %timeit g2(10)
The slowest run took 25.87 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 73.7 ns per loop
```

However, I don't think the 60ns speedup is worth the "huh?" reaction that the code invokes on reading.

@jonparrott @lukesneeringer WDYT?